Ntlm.cs

```diff
@@ -222,17 +224,29 @@ static bool IsExactlyOneBitSet(uint i)
             return i != 0 && (i & (i - 1)) == 0;
         }
 
-        public unsafe string CreateNegotiateMessage(bool spnego = false)
+        private unsafe void CreateNegotiateMessageRaw(Span<byte> asBytes)
         {
-            Debug.Assert(HeaderLenght == NtlmHeader.Length);
+            Debug.Assert(HeaderLength == NtlmHeader.Length);
+            Debug.Assert(asBytes.Length == sizeof(NegotiateMessage));
 
-            Span<byte> asBytes = stackalloc byte[sizeof(NegotiateMessage)];
             Span<NegotiateMessage> message = MemoryMarshal.Cast<byte, NegotiateMessage>(asBytes);
 
             asBytes.Clear();
             NtlmHeader.CopyTo(asBytes);
             message[0].Header.MessageType = MessageType.Negotiate;
-            message[0].Flags = Flags.NegotiateNtlm2 | Flags.NegotiateUnicode | Flags.TargetName | Flags.TargetTypeServer | Flags.NegotiateAlwaysSign;
+            message[0].Flags =
+                Flags.NegotiateNtlm2 | Flags.NegotiateNtlm | Flags.NegotiateUnicode | Flags.TargetName |
+                Flags.NegotiateVersion | Flags.NegotiateKeyExchange | Flags.Negotiate128 |
+                Flags.NegotiateTargetInfo | Flags.NegotiateAlwaysSign | Flags.NegotiateSign;
+            message[0].Version = new Version { VersionMajor = 6, VersionMinor = 1, ProductBuild = 7600, CurrentRevision = 15 };
```

Comment:
Do we need version? I don't mind but curious.

Reply:
I just tried to get the packet format as close as possible to other implementations.